tests/test_dxss.py

```diff
@@ -13,52 +13,26 @@ def test_dummy():
 
 def test_pardiso():
     """Test for PyPardiso. Print info for `pytest -s` logging."""
-    from dxss.solve_1d import PySolver, pypardiso
+    import dxss._solver_backend
 
-    if pypardiso is not None:
+    if dxss._solver_backend.pypardiso is not None:
         import pypardiso as _  # noqa: F401
 
         print("PyPardiso is available.")
     else:
         print("PyPardiso is not available.")
-        with pytest.warns(UserWarning):
-            PySolver(Asp=None, psolver=None)
 
 
-def test_mock_no_pypardiso_for_solve_1d(mocker):
+def test_mock_no_pypardiso_for_solver_warns(mocker):
     # mock no pypardiso installed (even on systems where it's installed)
     mocker.patch.dict(sys.modules, {"pypardiso": None})
-    importlib.reload(sys.modules["dxss.solve_1d"])
 
-    # check that trying to import raises an ImportError
-    with pytest.raises(ImportError):
-        import pypardiso
+    # need to reload the modules in case they where cached
+    import dxss._solvers  # fmt: skip
 
-    # now check that solve_1d gracefully handles missing pypardiso and warns
-    from dxss.solve_1d import PySolver, pypardiso  # noqa: F811
-
-    assert pypardiso is None
-    with pytest.warns(UserWarning):
-        PySolver(Asp=None, psolver=None)
-
-
-def test_mock_no_pypardiso_for_solve_2d(mocker):
-    mocker.patch.dict(sys.modules, {"pypardiso": None})
-    with pytest.raises(ImportError):
-        import pypardiso
-    from dxss.solve_2d import PySolver, pypardiso  # noqa: F811
+    importlib.reload(sys.modules["dxss._solver_backend"])
+    importlib.reload(sys.modules["dxss._solvers"])
 
-    assert pypardiso is None
-    with pytest.warns(UserWarning):
-        PySolver(Asp=None, psolver=None)
-
-
-def test_mock_no_pypardiso_for_solve_3d(mocker):
-    mocker.patch.dict(sys.modules, {"pypardiso": None})
-    with pytest.raises(ImportError):
-        import pypardiso
-    from dxss.solve_3d import PySolver, pypardiso  # noqa: F811
-
-    assert pypardiso is None
+    # now check that solve_1d gracefully handles missing pypardiso and warns
```

Comment:
```suggestion
    # now check that solver gracefully handles missing pypardiso and warns
```
As I think no longer specific to `solve_1d`